[Fleet] Update docker image of registry used in integration tests

jfsiii opened the pull request:
## Summary

 * Use a more recent Docker image for package registry used in integration/API tests. 
    * [New image](https://container-library.elastic.co/r/package-registry/distribution:fc104ac437370d80518e24da6d0b84370edf0c0c) is from Jun-14 [old image](https://container-library.elastic.co/r/package-registry/distribution:b6a53ac9300333a4a45f3f7d350c9aed72061a66) is from Apr-14
 * Add support for using a custom Docker registry using `FLEET_PACKAGE_REGISTRY_DOCKER_IMAGE` env var
    * e.g. `production` tag
    ```
    FLEET_PACKAGE_REGISTRY_DOCKER_IMAGE='docker.elastic.co/package-registry/distribution:production' FLEET_PACKAGE_REGISTRY_PORT=12345 yarn test:ftr:runner
    ```
    or a personal one
    ```
    FLEET_PACKAGE_REGISTRY_DOCKER_IMAGE='docker.elastic.co/employees/jfsiii/package-registry-distribution:kb90454' FLEET_PACKAGE_REGISTRY_PORT=12345 yarn test:ftr:runner

    ```

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130512/)
* Commit: d471ab31f50a602f9ae6f1ad669c568f630c1a59
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101911/d471ab31f50a602f9ae6f1ad669c568f630c1a59)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroupDocker](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130512/execution/node/768/log)
* [Execute xpack-kibana-ciGroupDocker](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130512/execution/node/821/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130512/","number":"130512","commit":"d471ab31f50a602f9ae6f1ad669c568f630c1a59"}]}
PIPELINE-->

jfsiii commented:
@elasticmachine merge upstream

ruflin commented on x-pack/test/fleet_api_integration/config.ts:
```diff
@@ -14,8 +14,7 @@ import { defineDockerServersConfig } from '@kbn/test';
 // It should be updated any time there is a new Docker image published for the Snapshot Distribution of the Package Registry.
-export const dockerImage =
-  'docker.elastic.co/package-registry/distribution:b6a53ac9300333a4a45f3f7d350c9aed72061a66';
+export const dockerImage = 'docker.elastic.co/package-registry/distribution:staging';
```
The fixed commit was introduced during the time when we had frequent changes. It made sure, any change to the registry will not break CI builds for all of Kibana. By now, things are much more stable. 

I like the idea of frequently updating it. Even though `staging`, more likely `production` should always work, I kind of like the fact that a CI build is always reproducible. The part I don't like is that we have to update this manually too often and catch bugs way too late.

We had similar issues for the stack versions for Beats / Agents builds. Now we have a script in place that on a predefined period updates the version hash with a pull request and it automatically gets merged when it passes. I wonder if this would bring us the best of both worlds?

jen-huang replied:
++ to above. @jfsiii As we discussed, can we update the image here to a newer version (but still using a fixed build) and get this PR in? We can then iterate on improving the process to update the image we use more frequently.

jfsiii replied:
@ruflin @jen-huang Updated to a newer version in https://github.com/elastic/kibana/pull/101911/commits/32a130cbd7662055d7fa1b8684c8d75a96a696af

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131476/)
* Commit: f3f7b2a8039ec8c956a11ab97775bb7c378d850a
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101911/f3f7b2a8039ec8c956a11ab97775bb7c378d850a)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroupDocker](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131476/execution/node/768/log)
* [Execute xpack-kibana-ciGroupDocker](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131476/execution/node/821/log)

### History
* :broken_heart: [Build #130512](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130512/) failed d471ab31f50a602f9ae6f1ad669c568f630c1a59

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130512/","number":"130512","commit":"d471ab31f50a602f9ae6f1ad669c568f630c1a59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131476/","number":"131476","commit":"f3f7b2a8039ec8c956a11ab97775bb7c378d850a"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131479/)
* Commit: a083d25c2315185ea62285855822809aaaa4c322
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101911/a083d25c2315185ea62285855822809aaaa4c322)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #131476](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131476/) failed f3f7b2a8039ec8c956a11ab97775bb7c378d850a
* :broken_heart: [Build #130512](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130512/) failed d471ab31f50a602f9ae6f1ad669c568f630c1a59

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130512/","number":"130512","commit":"d471ab31f50a602f9ae6f1ad669c568f630c1a59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131476/","number":"131476","commit":"f3f7b2a8039ec8c956a11ab97775bb7c378d850a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131479/","number":"131479","commit":"a083d25c2315185ea62285855822809aaaa4c322"}]}
PIPELINE-->

jfsiii commented on x-pack/test/fleet_api_integration/config.ts:
```diff
@@ -14,8 +14,7 @@ import { defineDockerServersConfig } from '@kbn/test';
 // It should be updated any time there is a new Docker image published for the Snapshot Distribution of the Package Registry.
-export const dockerImage =
-  'docker.elastic.co/package-registry/distribution:b6a53ac9300333a4a45f3f7d350c9aed72061a66';
+export const dockerImage = 'docker.elastic.co/package-registry/distribution:staging';
```
```suggestion
export const dockerImage = 'docker.elastic.co/package-registry/distribution:fc104ac437370d80518e24da6d0b84370edf0c0c';
```

jfsiii commented:
@elasticmachine merge upstream

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

jfsiii commented on x-pack/test/fleet_api_integration/config.ts:
```diff
@@ -14,8 +14,7 @@ import { defineDockerServersConfig } from '@kbn/test';
 // It should be updated any time there is a new Docker image published for the Snapshot Distribution of the Package Registry.
-export const dockerImage =
-  'docker.elastic.co/package-registry/distribution:b6a53ac9300333a4a45f3f7d350c9aed72061a66';
+export const dockerImage = 'docker.elastic.co/package-registry/distribution:fc104ac437370d80518e24da6d0b84370edf0c0c';
```
```suggestion
export const dockerImage =
  process.env.FLEET_PACKAGE_REGISTRY_DOCKER_IMAGE ||
  'docker.elastic.co/package-registry/distribution:fc104ac437370d80518e24da6d0b84370edf0c0c';
```

jfsiii commented:
@ruflin @jen-huang I updated this PR to include a change https://github.com/elastic/kibana/pull/101911/commits/ae88300db1a21e3a89a62b9c639d4e386b9a3c27 which allows people to override the default image with a specific one using an env variable 

I can drop it and move to a separate PR if you would like to discuss it, but I added it here to make things faster if it's a 👍🏻

jen-huang reviewed (approved):
Love the env var enhancement. 🚀 on happy CI

jen-huang commented on x-pack/test/fleet_api_integration/config.ts:
```diff
@@ -15,7 +15,8 @@ import { defineDockerServersConfig } from '@kbn/test';
 // It should be updated any time there is a new Docker image published for the Snapshot Distribution of the Package Registry.
 export const dockerImage =
-  'docker.elastic.co/package-registry/distribution:b6a53ac9300333a4a45f3f7d350c9aed72061a66';
+  process.env.FLEET_PACKAGE_REGISTRY_DOCKER_IMAGE ||
```
could you document this in our README? https://github.com/elastic/kibana/blob/master/x-pack/plugins/fleet/README.md#api-integration-tests

jfsiii replied:
Added in https://github.com/elastic/kibana/pull/101911/commits/527fdef8fda495c4ec9237b62ff0624a4d50e9c1

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/132053/)
* Commit: ae88300db1a21e3a89a62b9c639d4e386b9a3c27
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101911/ae88300db1a21e3a89a62b9c639d4e386b9a3c27)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #131479](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131479/) succeeded a083d25c2315185ea62285855822809aaaa4c322
* :broken_heart: [Build #131476](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131476/) failed f3f7b2a8039ec8c956a11ab97775bb7c378d850a
* :broken_heart: [Build #130512](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130512/) failed d471ab31f50a602f9ae6f1ad669c568f630c1a59

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @jfsiii

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130512/","number":"130512","commit":"d471ab31f50a602f9ae6f1ad669c568f630c1a59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131476/","number":"131476","commit":"f3f7b2a8039ec8c956a11ab97775bb7c378d850a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131479/","number":"131479","commit":"a083d25c2315185ea62285855822809aaaa4c322"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/132053/","number":"132053","commit":"ae88300db1a21e3a89a62b9c639d4e386b9a3c27"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/132146/)
* Commit: 527fdef8fda495c4ec9237b62ff0624a4d50e9c1
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101911/527fdef8fda495c4ec9237b62ff0624a4d50e9c1)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #132053](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/132053/) succeeded ae88300db1a21e3a89a62b9c639d4e386b9a3c27
* :green_heart: [Build #131479](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131479/) succeeded a083d25c2315185ea62285855822809aaaa4c322
* :broken_heart: [Build #131476](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131476/) failed f3f7b2a8039ec8c956a11ab97775bb7c378d850a
* :broken_heart: [Build #130512](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130512/) failed d471ab31f50a602f9ae6f1ad669c568f630c1a59

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @jfsiii

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130512/","number":"130512","commit":"d471ab31f50a602f9ae6f1ad669c568f630c1a59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131476/","number":"131476","commit":"f3f7b2a8039ec8c956a11ab97775bb7c378d850a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131479/","number":"131479","commit":"a083d25c2315185ea62285855822809aaaa4c322"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/132053/","number":"132053","commit":"ae88300db1a21e3a89a62b9c639d4e386b9a3c27"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/132146/","number":"132146","commit":"527fdef8fda495c4ec9237b62ff0624a4d50e9c1"}]}
PIPELINE-->

kibanamachine commented:
## 💚 Backport successful

| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [7.x](https://github.com/elastic/kibana/pull/102560)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/102560">](https://github.com/elastic/kibana/pull/102560) |

This backport PR will be merged automatically after passing CI.

ruflin commented:
Being able to use an environment variable to change the version of the registry is great! Thanks.
